client/components/test-mode-notice/test/index.js

```diff
@@ -92,89 +91,15 @@ describe( 'Test mode notification', () => {
 		}
 	);
 
-	test.each( allTopics )( 'Returns correct notice', ( topic ) => {
-		const expected = (
-			<Notice
-				className="wcpay-test-mode-notice"
-				status="warning"
-				isDismissible={ false }
-			>
-				{ getNoticeMessage( topic ) }
-			</Notice>
-		);
-
-		expect( getNotice( topic ) ).toStrictEqual( expected );
-	} );
-
 	test.each( topicsWithTestMode )(
-		'Simple component is wrapped correctly',
+		'Component is rendered correctly according to test mode',
```

Comment:
I think `Component is rendered correctly` is a better description here 🙂 . The "according to test mode" is implied in stating that the component renders correctly.